packages/actions/src/utils/toChecksumAddress.js

```
@@ -0,0 +1,42 @@
+    // @ts-ignore
+    const shouldUpperCase = Number.parseInt(hash[i], 16) > 7;
+
+    if (charCode >= 48 && charCode <= 57) {  // '0'-'9'
+      return char;  // Numbers remain unchanged
+    } else if (charCode >= 97 && charCode <= 102) {  // 'a'-'f'
+      return shouldUpperCase ? char.toUpperCase() : char;
+    } else {
+      throw new Error(`Invalid character ${char} in address.`);
+    }
+  })
+  .join('');
```

Biome

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)